Hadrons/Modules/MContraction/Baryon.hpp

```diff
@@ -135,23 +135,49 @@ void TBaryon<FImpl1, FImpl2,FImpl3>::parseGammaString(std::vector<GammaABPair> &
     
     std::string gammaString = par().gammas;
     //Shorthands for standard baryon operators
-    gammaString = regex_replace(gammaString, std::regex("j12"),"Identity SigmaXZ");
-    gammaString = regex_replace(gammaString, std::regex("j32X"),"Identity GammaZGamma5");
-    gammaString = regex_replace(gammaString, std::regex("j32Y"),"Identity GammaT");
-    gammaString = regex_replace(gammaString, std::regex("j32Z"),"Identity GammaXGamma5");
+    gammaString = regex_replace(gammaString, std::regex("j12"),"(Identity SigmaXZ)");
+    gammaString = regex_replace(gammaString, std::regex("j32X"),"(Identity GammaZGamma5)");
+    gammaString = regex_replace(gammaString, std::regex("j32Y"),"(Identity GammaT)");
+    gammaString = regex_replace(gammaString, std::regex("j32Z"),"(Identity GammaXGamma5)");
     //Shorthands for less common baryon operators
-    gammaString = regex_replace(gammaString, std::regex("j12_alt1"),"Gamma5 SigmaYT");
-    gammaString = regex_replace(gammaString, std::regex("j12_alt2"),"Identity GammaYGamma5");
-  
-    std::vector<GammaAB> gamma_help;
-    gamma_help = strToVec<GammaAB>(gammaString);
-    LOG(Message) << gamma_help.size() << " " << gamma_help.size()%2 << std::endl;
-    assert(gamma_help.size()%2==0 && "need even number of gamma-pairs.");
-    gammaList.resize(gamma_help.size()/2);
-
-    for (int i = 0; i < gamma_help.size()/2; i++){
-        gammaList[i].first=gamma_help[2*i];
-        gammaList[i].second=gamma_help[2*i+1];
+    gammaString = regex_replace(gammaString, std::regex("j12_alt1"),"(Gamma5 SigmaYT)");
+    gammaString = regex_replace(gammaString, std::regex("j12_alt2"),"(Identity GammaYGamma5)");
+    
+    //A single gamma matrix 
+    std::regex rex_g("([0-9a-zA-Z]+)");
+    //The full string we expect
+    std::regex rex("( *\\(( *\\(([0-9a-zA-Z]+) +([0-9a-zA-Z]+) *\\)){2} *\\) *)+");
+    std::smatch sm;
+    std::regex_match(gammaString, sm, rex);
+    assert(sm[0].matched && "invalid gamma structure.");
+
```

Comment:
regex_match does not allow to iterate through tokens - it only finds the first matches of each regular expression! this is why I used sregex_iterator afterwards